app/Services/QueryServiceProvider.js

```diff
@@ -37,6 +50,202 @@ async function getUnresolvedVocabulary(languagePackageId, userId) {
+          attributes: ['active'],
+          required: true,
+        },
+      ],
+      order: Sequelize.literal('random()'),
+      limit: vocabularyLimit,
+      attributes: ['id', 'name'],
+      where: {
+        drawerId: drawer.id,
+        lastQuery: { [Op.lt]: queryDate },
+        '$Group.active$': true,
+        active: true,
+      },
+    });
+
+    vocabs.push(...vocabularies);
+  }
+  /* eslint-enable no-await-in-loop */
+
+  return vocabs.map((vocab) => deleteKeysFromObject(['Group'], vocab.toJSON()));
+}
+
+// return the unactivated vocabulary
+async function getUnactivatedVocabulary(languagePackageId, userId) {
+  // Get drawers id
+  const drawer = await Drawer.findOne({
+    attributes: ['id'],
+    where: {
+      userId,
+      languagePackageId,
+      stage: 0,
+    },
+  });
+
+  // return every vocabulary in drawer 0
+  const vocabularies = await VocabularyCard.findAll({
+    include: [
+      {
+        model: Translation,
+        attributes: ['name'],
+      },
+      {
+        model: Group,
+        attributes: ['active'],
+        required: true,
+      },
+    ],
+    order: Sequelize.literal('random()'),
+    attributes: ['id', 'name'],
+    where: {
+      drawerId: drawer.id,
+      '$Group.active$': true,
+      active: true,
+    },
+  });
+  return vocabularies.map((vocab) => deleteKeysFromObject(['Group'], vocab.toJSON()));
+}
+
+// function to handle correct query
+async function handleCorrectQuery(userId, vocabularyId) {
+  // fetch selected vocabulary card
+  const vocabularyCard = await VocabularyCard.findOne({
+    include: [
+      {
+        model: Drawer,
+        attributes: ['stage'],
+      },
+    ],
+    attributes: ['id', 'name', 'drawerId', 'languagePackageId'],
+    where: {
+      userId,
+      id: vocabularyId,
+    },
+  });
+
+  // push vocabulary card one drawer up
+  // get drawer id from stage
+
+  const drawer = await Drawer.findOne({
+    attributes: ['id'],
+    where: {
+      userId,
+      languagePackageId: vocabularyCard.languagePackageId,
+      stage: vocabularyCard.Drawer.stage + 1,
+    },
+  });
+  if (!drawer) {
+    // if no output there is no next drawer => stop
+    return;
+  }
+
+  // update drawerId for vocabulary card
+  vocabularyCard.lastQuery = new Date();
+  vocabularyCard.drawerId = drawer.id;
+
+  await vocabularyCard.save();
+}
+
+// function to handle wrong query
+async function handleWrongQuery(userId, vocabularyId) {
+  // if query was solved wrong, push vocabulary card in drawer one
+  const vocabularyCard = await VocabularyCard.findOne({
+    include: [
+      {
+        model: Drawer,
+        attributes: ['stage'],
+      },
+    ],
+    attributes: ['id', 'name', 'drawerId', 'languagePackageId'],
+    where: {
+      userId,
+      id: vocabularyId,
+    },
+  });
+
+  const drawer = await Drawer.findOne({
+    attributes: ['id'],
+    where: {
+      userId,
+      languagePackageId: vocabularyCard.languagePackageId,
+      stage: 1,
```

Comment:
wrong queries from stage 0 move the card into the first drawer. shouldn't we add some logic to obmit this behavior?